src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/indexes/RegularIndexDefinitionDesc.java

```diff
@@ -15,17 +30,117 @@
   TableDescConstants.IndexDefinitionDesc.OPTIONS
 })
 public record RegularIndexDefinitionDesc(
+
+    /*
+     * API regular column can be primitive or map/set/list collection type.
+     * CreateIndexCommand on collection column will have index function.
+     * User can specify or use the default index function.
+     * Details see {@link RegularIndexColumnDeserializer}
+     */
     @NotNull
-        @Schema(description = "Name of the column to index.", required = true)
+        @Schema(
+            description =
+                "Name of the column to index. (Optional: can also specify the index function for map column. E.G. {\"column\": {\"mapColumn\" : \"$keys\"}}",
+            required = true)
+        @JsonDeserialize(using = RegularIndexColumnDeserializer.class)
+        @JsonSerialize(using = RegularIndexColumnSerializer.class)
         @JsonProperty(TableDescConstants.IndexDefinitionDesc.COLUMN)
-        String column,
-    //
+        RegularIndexColumn column,
     @JsonInclude(JsonInclude.Include.NON_NULL)
         @Nullable
         @Schema(description = "Indexing options.", type = SchemaType.OBJECT)
```

Comment:
Update to "Options for the new index, not all options are valid for all columns. Check documentation for details"